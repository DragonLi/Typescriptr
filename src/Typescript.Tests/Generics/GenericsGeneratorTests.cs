using System;
using Assent;
using Typescriptr;
using Typescriptr.Formatters;
using Xunit;
using Xunit.Abstractions;

namespace Typescript.Tests.Generics
{
    public class GenericsGeneratorTests
    {
        class Alpha { }
        class Beta { }

        class BaseType<T1, T2>
        {
        }

        class TypeWithGenericArguments : BaseType<Alpha, Beta>
        {
        }

        [Fact]
        public void Generator_TypeWithGenericArguments_GeneratesSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithGenericArguments) });

            var result = generated.Types;

            this.Assent(result);
        }

        class TypeWithOpenGenericArguments<T>
        {
        }

        [Fact]
        public void Generator_TypeWithOpenGenericArguments_GeneratesSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] { typeof(TypeWithOpenGenericArguments<>) });

            var result = generated.Types;

            this.Assent(result);
        }

    }
}