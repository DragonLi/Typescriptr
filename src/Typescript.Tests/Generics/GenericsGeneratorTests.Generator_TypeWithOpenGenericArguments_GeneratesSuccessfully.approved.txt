declare namespace Api {
  interface TypeWithOpenGenericArguments<T> {
  }
}
